index.js

```diff
@@ -237,6 +237,37 @@ const execa = (command, args, options) => {
 		return Promise.reject(error);
 	}
 
+	const originalKill = spawned.kill.bind(spawned);
+	spawned.kill = (signal, options) => {
+		if (signal === undefined ||
+			(typeof signal === 'string' && signal.toUpperCase() === 'SIGTERM') ||
+			(typeof signal === 'number' && signal === os.constants.signals.SIGTERM)) {
+			const retryAfter = options !== undefined &&
+				['number', 'string'].includes(typeof options.retryAfter) ?
+				Number(options.retryAfter) :
+				5000;
+
+			const retry = !(options !== undefined && options.retry === false);
+
+			const killResult = originalKill(signal);
+
+			if (retry && killResult) {
+				const safetyTimeout = setTimeout(() => {
+					try {
+						originalKill('SIGKILL');
+					} catch (_) {
+						// All good - process killed
+					}
+				}, retryAfter);
+				safetyTimeout.unref();
+			}
+
+			return killResult;
+		}
+
+		return originalKill(signal);
```

Comment:
Some style suggestion: could this line be merged with the `originalKill()` from above? This is possible by moving the `if` blocks around.

Reply:
Oh yes, thanks: c86bb8ab10b0e4a483678130eec92df4f525c6c0